src/py21cmfast/wrapper.py

```diff
@@ -3219,3 +3223,308 @@ def calibrate_photon_cons(
             nf_estimate=neutral_fraction_photon_cons,
             NSpline=len(z_for_photon_cons),
         )
+
+
+def run_kSZ(
+    lc=None,
+    z_start=5,
+    z_end=15,
+    cosmo_params=None,
+    user_params=None,
+    PARALLEL_APPROX=False,
+    rotation=True,
+    random_seed=1,
+):
+    r"""
+    Run the kSZ module.
+
+    The function takes the cosmological quantities used in the lightcone when applicable.
+
+
+    Parameters
+    ----------
+    folder: str
+        Folder name where everything is.
+    z_start : float, optional
+        Starting redshift for kSZ calculation, default is 5.
+    z_end : float, optional
+        Ending redshift for kSZ calculation, default is 15.
+    user_params : `~UserParams`, optional
+        Defines the overall options and parameters of the run.
+    cosmo_params : :class:`~CosmoParams`, optional
+        Defines the cosmological parameters used to compute initial conditions.
+    PARALLEL_APPROX : bool, optional
+        Flag for parrallel approximation, if True, parallel approximation is taken which is much quicker, but more approximate. Default: False.
+    rotation : bool, optional
+        Flag for rotation of boxes, if True boxes are shifted for every HII_DIM, which is the size of the simulation. Default: True.
+
+    Returns
+    -------
+    kSZ_output :
+        KSZ_box: map of the kSZ effect, in Kelvins.
+        taue_boxp: map of the optical depth.
+    """
+    if lc is None:
+        if user_params is None:
+            user_params = UserParams(**UserParams._defaults_)
+        if cosmo_params is None:
+            cosmo_params = CosmoParams(**CosmoParams._defaults_)
+        lc_quantities = ("brightness_temp", "xH_box", "density", "velocity_z")
+        lc = run_lightcone(
+            redshift=z_start,
+            user_params=user_params,
+            cosmo_params=cosmo_params,
+            lightcone_quantities=lc_quantities,
+            random_seed=random_seed,
+        )
+    random.seed(random_seed)
+
+    kSZ_consts = KszConstants(
```

Comment:
if define all functions in run_kSZ, then this constants class thing can be put in _int